src/components/Charts/Legend/Legend.js

```diff
@@ -1,266 +1,385 @@
+          .style('font-size', '10px')
+
+        var legendOrdinal = legendColor()
+          .orient('horizontal')
+          .shapePadding(self.LEGEND_WIDTH)
+          .labelWrap(this.MAX_LEGEND_WIDTH)
+          .labelAlign('left')
+          .scale(legendScale)
+          .on('cellclick', function (d) {
+            self.onClick(d, legendLabels)
+          })
+
+        select(this.bottomLegendElement).call(legendOrdinal).style('font-family', 'inherit')
+      }
+
+      this.applyStylesForHiddenSeries(legendElement, legendLabels)
+    } catch (error) {
+      console.error(error)
+    }
   }
 
-  legend.textWrap = function (_) {
-    if (!arguments.length) {return textWrap}
-    textWrap = _
-    return legend
+  applyStylesForHiddenSeries = (legendElement, legendLabels) => {
+    try {
+      select(legendElement)
+        .selectAll('.cell')
+        .each(function (label) {
+          const legendLabel = legendLabels.find((l) => l.label === label)
+          if (legendLabel) {
+            select(this).select('.swatch').attr('stroke', legendLabel.color).attr('stroke-location', 'outside')
+            if (legendLabel.hidden) {
+              select(this).attr('class', 'cell hidden')
+            } else {
+              select(this).attr('class', 'cell visible')
+            }
+          }
+        })
+    } catch (error) {
+      console.error(error)
+    }
   }
 
-  legend.on = function () {
-    const value = legendDispatcher.on.apply(legendDispatcher, arguments)
-    return value === legendDispatcher ? legend : value
+  getLegendScale = (legendLabels) => {
+    const colorRange = legendLabels.map((obj) => {
+      return obj.color
+    })
+
+    return scaleOrdinal()
+      .domain(
+        legendLabels.map((obj) => {
+          return obj.label
+        }),
+      )
+      .range(colorRange)
   }
 
-  return legend
+  render = () => {
+    return (
+      <g data-test='legend'>
+        {this.props.placement === 'right' && (
+          <>
+            <g
+              ref={(el) => (this.rightLegendElement = el)}
+              id={this.LEGEND_ID}
+              data-test='right-legend'
+              className='legendOrdinal right-legend'
+              transform={`translate(${this.LEFT_PADDING}, 8)`}
+            >
+              {/* {this.props.legendColumn && (
+              <LegendSelector
+                {...this.props}
+                column={this.props.legendColumn}
+                positions={['bottom']}
+                align='end'
+                childProps={{
+                  ref: (r) => (this.legendBorder = r),
+                  x: (this.titleBBox?.x ?? 0),
+                  y: (this.titleBBox?.y ?? 0),
+                  width: (this.titleBBox?.width ?? 0) + this.BUTTON_PADDING * 2,
+                  height: (this.titleBBox?.height ?? 0) + this.BUTTON_PADDING * 2,
+                  transform: this.props.translate,
+                }}
+              />
+            )} */}
```

Comment:
cleanup?

Reply:
This one I'll leave in but I left a note